src/lib/core/sio.c

```diff
@@ -245,7 +281,9 @@ sio_bind(int fd, const struct sockaddr *addr, socklen_t addrlen)
 {
 	int rc = bind(fd, addr, addrlen);
 	if (rc < 0)
-		diag_set(SocketError, sio_socketname(fd), "bind");
+		diag_set(SocketError,
+			 sio_socketname_addr(fd, addr, addrlen, NULL, 0),
```

Comment:
Please, let's not pass NULL as peers. If we fail to bind to an address, e.g. due to already taken socket, we probably wanna see, which peer already taken that address. Let's move getting peer name to `sio_socketname_addr`

Reply:
Ok. Fixed